app/bundles/FormBundle/Controller/PublicController.php

```diff
@@ -402,18 +402,28 @@ public function embedAction(Request $request)
      */
     private function replacePostSubmitTokens($string, SubmissionEvent $submissionEvent): string|array
     {
-        if (empty($this->tokens)) {
-            if ($lead = $submissionEvent->getLead()) {
-                $this->tokens = array_merge(
-                    $submissionEvent->getTokens(),
-                    TokenHelper::findLeadTokens(
-                        $string,
-                        $lead->getProfileFields()
-                    )
-                );
-            }
+        if (count($this->tokens)) {
```

Comment:
I like that you "return early" but why did you choose to use `count()` instead of `empty()` which used to be there? I know this is a micro-optimization but `empty()` [is faster](https://stackoverflow.com/questions/2216110/checking-for-empty-arrays-count-vs-empty).

Reply:
Main reason was reading somewhere that it is best to avoid checking something with negations like `!empty`.

Thus, I try to use “positive” if statements.

This approach resonates with me, because in most languages avoiding `not` makes sentences clearer

- We should not turn left, versus
- We should turn right

The latter is immediately clear. That's the way I see if statements. But yeah, sometimes it might not be optimal. 

That's my 2 cents RE my reasoning :sweat_smile:  What are your thoughts about that?

Reply:
That's definitely valid reasoning! Thank you for sharing! Now it's performance vs readability. A micro-improvement for both in this case. I'm used to read "not empty" whether it is `!empty($array)` or just `!$array` but I get it can be different for others with different backgrounds. I'm OK with both.